internal/dag/gatewayapi_processor.go

```diff
@@ -150,15 +152,47 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 				if gatewayAllowMatches {
 					// Empty Selector matches all routes.
-					matchingRoutes = append(matchingRoutes, route)
+					matchingHTTPRoutes = append(matchingHTTPRoutes, route)
 				}
 			}
 		}
 
-		// Process all the routes that match this Gateway.
-		for _, matchingRoute := range matchingRoutes {
+		for _, route := range p.source.tlsroutes {
+			// Filter the TLSRoutes that match the gateway which Contour is configured to watch.
+			// RouteBindingSelector defines a schema for associating routes with the Gateway.
+			// If Namespaces and Selector are defined, only routes matching both selectors are associated with the Gateway.
+
+			// ## RouteBindingSelector ##
+			//
+			// Selector specifies a set of route labels used for selecting routes to associate
+			// with the Gateway. If this Selector is defined, only routes matching the Selector
+			// are associated with the Gateway. An empty Selector matches all routes.
+
+			nsMatches, err := p.namespaceMatches(listener.Routes.Namespaces, route.Namespace)
```

Comment:
should we add some route matching/namespace matching tests for tlsroutes as well?

Reply:
I opened https://github.com/projectcontour/contour/pull/3684 to cover tests. Was trying to reduce the review sizes and was already there (also found a bug). =)